internal/pkg/artifact/download/http/downloader.go

```diff
@@ -156,10 +172,155 @@ func (e *Downloader) downloadFile(ctx context.Context, artifactName, filename, f
+	n := len(b)
+	dp.downloaded.Add(n)
+	return n, nil
+}
+
+func (dp *downloadProgressReporter) Report(ctx context.Context) {
+	started := time.Now()
+	dp.started = started
+	sourceURI := dp.sourceURI
+	log := dp.log
+	length := dp.length
+	warnTimeout := dp.warnTimeout
+	interval := dp.interval
+
+	go func() {
+		t := time.NewTimer(interval)
+		defer t.Stop()
+		for {
+			select {
+			case <-ctx.Done():
+				return
+			case <-t.C:
+				now := time.Now()
+				timePast := now.Sub(started)
+				downloaded := float64(dp.downloaded.Load())
+				bytesPerSecond := downloaded / float64(timePast/time.Second)
+
+				var msg string
+				var args []interface{}
+				if length > 0 {
+					// length of the download is known, so more detail can be provided
+					percentComplete := downloaded / length * 100.0
+					msg = "download progress from %s is %s/%s (%.2f%% complete) @ %sps"
+					args = []interface{}{
+						sourceURI, units.HumanSize(downloaded), units.HumanSize(length), percentComplete, units.HumanSize(bytesPerSecond),
+					}
+				} else {
+					// length unknown so provide the amount downloaded and the speed
+					msg = "download progress from %s has fetched %s @ %sps"
+					args = []interface{}{
+						sourceURI, units.HumanSize(downloaded), units.HumanSize(bytesPerSecond),
+					}
+				}
+
+				log.Infof(msg, args...)
+				if timePast >= warnTimeout {
+					// duplicate to warn when over the warnTimeout; this still has it logging to info that way if
+					// they are filtering the logs to info they still see the messages when over the warnTimeout, but
+					// when filtering only by warn they see these messages only
+					log.Warnf(msg, args...)
+				}
+			}
+		}
+	}()
+}
+
+func (dp *downloadProgressReporter) ReportComplete() {
+	now := time.Now()
+	timePast := now.Sub(dp.started)
+	downloaded := float64(dp.downloaded.Load())
+	bytesPerSecond := downloaded / float64(timePast/time.Second)
+	msg := "download from %s completed in %s @ %sps"
+	args := []interface{}{
+		dp.sourceURI, units.HumanDuration(timePast), units.HumanSize(bytesPerSecond),
+	}
+	dp.log.Infof(msg, args...)
+	if timePast >= dp.warnTimeout {
+		// see reason in `Report`
+		dp.log.Warnf(msg, args...)
+	}
+}
+
+func (dp *downloadProgressReporter) ReportFailed(err error) {
+	now := time.Now()
+	timePast := now.Sub(dp.started)
+	downloaded := float64(dp.downloaded.Load())
+	bytesPerSecond := downloaded / float64(timePast/time.Second)
+	var msg string
+	var args []interface{}
+	if dp.length > 0 {
+		// length of the download is known, so more detail can be provided
+		percentComplete := downloaded / dp.length * 100.0
+		msg = "download from %s failed at %s/%s (%.2f%% complete) @ %sps - %s"
+		args = []interface{}{
+			dp.sourceURI, units.HumanSize(downloaded), units.HumanSize(dp.length), percentComplete, units.HumanSize(bytesPerSecond), err,
+		}
+	} else {
+		// length unknown so provide the amount downloaded and the speed
+		msg = "download from %s failed at %s @ %sps - %s"
```

Comment:
[NIT | Suggestion]
I find it more clear, even more "go-like" to have the error cause after a `:`.
```suggestion
		msg = "download from %s failed at %s @ %sps: %s"
```